atc/db/job_test.go

```diff
@@ -2109,186 +2109,447 @@ var _ = Describe("Job", func() {
-								Get:      "other-job-resource",
-								Resource: "some-resource",
+						Name:       "some-input",
+						JobID:      inputsJob.ID(),
+						ResourceID: someResource.ID(),
+						Passed: db.JobSet{
+							job1.ID(): true,
+							job2.ID(): true,
+						},
+						UseEveryVersion: true,
+						Trigger:         true,
+					},
+				}))
+			})
+		})
+
+		Context("when the input is pinned through the job config", func() {
+			BeforeEach(func() {
+				var err error
+				inputsPipeline, _, err = team.SavePipeline("inputs-pipeline", atc.Config{
+					Jobs: atc.JobConfigs{
+						{
+							Name: "some-job",
+							Plan: atc.PlanSequence{
+								{
+									Get:      "some-pinned-input",
+									Resource: "some-resource",
+									Version:  &atc.VersionConfig{Pinned: atc.Version{"input": "pinned"}},
+								},
 							},
 						},
 					},
-					{
-						Name: "job-1",
+					Resources: atc.ResourceConfigs{
+						{
+							Name:   "some-resource",
+							Type:   "some-type",
+							Source: atc.Source{"some": "source"},
+						},
 					},
+				}, db.ConfigVersion(0), false)
+				Expect(err).ToNot(HaveOccurred())
+
+				var found bool
+				inputsJob, found, err = inputsPipeline.Job("some-job")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(found).To(BeTrue())
+			})
+
+			It("pins the inputs to that version", func() {
+				someResource, found, err := inputsPipeline.Resource("some-resource")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(found).To(BeTrue())
+
+				Expect(inputs).To(Equal(db.InputConfigs{
 					{
-						Name: "job-2",
+						Name:          "some-pinned-input",
+						JobID:         inputsJob.ID(),
+						ResourceID:    someResource.ID(),
+						PinnedVersion: atc.Version{"input": "pinned"},
 					},
-				},
-				Resources: atc.ResourceConfigs{
-					{
-						Name: "some-resource",
-						Type: "some-type",
+				}))
+			})
+
+			Context("when the input is also pinned through the api", func() {
+				BeforeEach(func() {
+					pinnedResource, found, err := inputsPipeline.Resource("some-resource")
+					Expect(err).ToNot(HaveOccurred())
+					Expect(found).To(BeTrue())
+
+					setupTx, err := dbConn.Begin()
+					Expect(err).ToNot(HaveOccurred())
+
+					brt := db.BaseResourceType{
+						Name: "some-type",
+					}
+
+					_, err = brt.FindOrCreate(setupTx, false)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(setupTx.Commit()).To(Succeed())
+
+					resourceConfigScope, err := pinnedResource.SetResourceConfig(atc.Source{"some": "source"}, atc.VersionedResourceTypes{})
+					Expect(err).ToNot(HaveOccurred())
+
+					err = resourceConfigScope.SaveVersions([]atc.Version{
+						{"api": "pinned"},
+					})
+					Expect(err).NotTo(HaveOccurred())
+
+					versionID, found, err := pinnedResource.ResourceConfigVersionID(atc.Version{"api": "pinned"})
+					Expect(err).NotTo(HaveOccurred())
+					Expect(found).To(BeTrue())
+
+					pinned, err := pinnedResource.PinVersion(versionID)
+					Expect(err).NotTo(HaveOccurred())
```

Comment:
Wait, I thought this pinning a version through the API if it's already pinned in the config returned an error - https://github.com/concourse/concourse/pull/5420/files#diff-37e62c267e080752855dde75c8aa7d8eR602-R604

Reply:
Ohh the test is for pinning through the get step. You can either pin the resource through it's resource config and also through the get step. Hm there might be a few things I should reword to make that more clear. I think I should rephrase the test to be more specific and possibly have it be `pinnedThroughResourceConfig` rather than `pinnedThroughConfig`? I might look around at rewording more things too to make a bigger distinction?

Reply:
Was not aware of that distinction! Thanks!